etcdserver/auth/auth.go

```diff
@@ -192,6 +182,10 @@ func (s *Store) CreateOrUpdateUser(user User) (out User, created bool, err error
 
 func (s *Store) CreateUser(user User) (User, error) {
 	u, err := s.createUserInternal(user)
+	// Attach root role to root user.
+	if u.User == "root" {
```

Comment:
I do not care about the logic you mentioned, since they are almost identical.

What I want to achieve here is to return the user that you created rather than returned a user that you modified after the creation.